src/cpu/x64/jit_uni_binary.cpp

```diff
@@ -46,8 +47,7 @@ static bool compare_layouts(const memory_desc_wrapper &src0_md,
     if (is_bcast) return true;
 
     bool same_layouts = true;
-    for (int d = 0; d < ndims; ++d)
-        same_layouts = same_layouts && strides0[d] == strides1[d];
+    same_layouts = std::equal(strides0, strides0 + ndims, strides1);
```

Comment:
I like this, and I don't know why they didn't use it in the first place. Note that `utils::array_cmp` also exists.